File: 1. SOLUCION DE ECUACIONES/6. Aceleracion de convergencia/6.1 Metodo de Aitken/aitken.c
#include <stdio.h>
#include <math.h>

double g(double n){
	/* === DEFINE THE FUCTION TO USE ===*/
	return 0;
}

double aitken(double p_0){
	double p_1 = g(p_0);
	double p_2 = g(p_0 + 1);
	double p_3 = g(p_0 + 2);

	double p = p_1 - (pow(p_2 - p_1, 2) / (p_3 - 2*p_2 + p_1));

	return p;
}

int main(){
	/* === INSERT DATA HERE === */
	double p_0;

	/* === CALCULATE === */
	double r = aitken(p_0);
	printf("%lf\n", r);
	return 0;
}

File: 1. SOLUCION DE ECUACIONES/6. Aceleracion de convergencia/6.1 Metodo de Aitken/aitken.m
function p = aitken(g, p_0)
		p_1 = g(p_0);
		p_2 = g(p_0 + 1);
		p_3 = g(p_0 + 2);
		p = p_1 - (power(p_2 - p_1, 2) / (p_3 - 2*p_2 + p_1) );
endfunction
